### ISSUE TITLE
:nest doesn't seem to work as documented

### ISSUE THREAD

**Issue by valerauko** [jkk/honeysql#370]
The README suggests that `:nest` could be used to explicitly wrap conditions in a WHERE in parens. It doesn't seem to work in certain conditions.

Discussion [thread on Clojurians Slack](https://clojurians.slack.com/archives/C66EM8D5H/p1636037757002200)

**Comment by seancorfield** [comment]
```clojure
dev=> (sql/format {:select :*
 #_=>  :from :vblog_posts
 #_=>  :where [:and
 #_=>          [:nest [:between :post_id 23 2105]]
 #_=>          [:not [:and
 #_=>                 [:or
 #_=>                  [:like :post_main "%[img%"]
 #_=>                  [:like :post_long "%[img%"]]
 #_=>                 [:< :post_date 1239462000]]]
 #_=>          [:not [:or
 #_=>                 [:like :post_main "%[devfav%"]
 #_=>                 [:like :post_long "%[devfav%"]]]
 #_=>          [:= :post_id 415]]
 #_=>  :limit 1})
["SELECT * FROM vblog_posts WHERE post_id BETWEEN ? AND ? AND NOT ((post_main LIKE ?) OR (post_long LIKE ?)) AND (post_date < ?) AND NOT (post_main LIKE ?) OR (post_long LIKE ?) AND (post_id = ?) LIMIT ?" 23 2105 "%[img%" "%[img%" 1239462000 "%[devfav%" "%[devfav%" 415 1]
dev=> (sql/format {:select :*
 #_=>  :from :vblog_posts
 #_=>  :where [:and
 #_=>          [[:nest [:between :post_id 23 2105]]]
 #_=>          [:not [:and
 #_=>                 [:or
 #_=>                  [:like :post_main "%[img%"]
 #_=>                  [:like :post_long "%[img%"]]
 #_=>                 [:< :post_date 1239462000]]]
 #_=>          [:not [:or
 #_=>                 [:like :post_main "%[devfav%"]
 #_=>                 [:like :post_long "%[devfav%"]]]
 #_=>          [:= :post_id 415]]
 #_=>  :limit 1})
["SELECT * FROM vblog_posts WHERE (post_id BETWEEN ? AND ?) AND NOT ((post_main LIKE ?) OR (post_long LIKE ?)) AND (post_date < ?) AND NOT (post_main LIKE ?) OR (post_long LIKE ?) AND (post_id = ?) LIMIT ?" 23 2105 "%[img%" "%[img%" 1239462000 "%[devfav%" "%[devfav%" 415 1]
```

**Comment by valerauko** [comment]
If the double `[[]]` is okay then I guess it'd do just documenting that in the readme with an example

**Comment by seancorfield** [comment]
I'm not sure why it behaves that way -- I'm not sure whether that's "expected" behavior (that just needs documentation) or not (in which case there's a bug).

**Comment by seancorfield** [comment]
It was a bug. It's fixed in 2.1.829.

**Comment by valerauko** [comment]
Thanks!
